hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/IncentiveDifferenceService.kt

```diff
@@ -0,0 +1,167 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.indexer.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.microsoft.applicationinsights.TelemetryClient
+import org.apache.commons.codec.binary.Base64
+import org.apache.commons.lang3.builder.Diff
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.util.DigestUtils
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.CurrentIncentive
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.DiffCategory
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.DiffableProperty
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.config.DiffProperties
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.config.TelemetryEvents
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.config.trackEvent
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.config.trackPrisonerEvent
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.repository.IncentiveHashRepository
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.services.events.HmppsDomainEventEmitter
+import java.time.Instant
+import kotlin.reflect.full.findAnnotations
+
```

Comment:
A lot of duplication here which hopefully can be refactored. Ive not done the unit test for this yet